src/viam/app/app_client.py

```diff
@@ -333,28 +494,27 @@ async def get_location(self, location_id: Optional[str] = None) -> Location:
 
         Args:
             location_id (Optional[str]): ID of the location to get. Defaults to the location ID provided at `AppClient` instantiation.
-                If no default location ID was passed, a GRPCError will be thrown.
 
         Raises:
-            GRPCError: If an invalid location ID is passed.
+            GRPCError: If an invalid location ID is passed or if one isn't passed and there was no location ID provided at `AppClient`
+                instantiation.
 
         Returns:
             viam.proto.app.Location: The location.
         """
-        location_id = location_id if location_id else self._location_id
-        request = GetLocationRequest(location_id=location_id if location_id else "")
+        request = GetLocationRequest(location_id=location_id if location_id else self._location_id if self._location_id else "")
         response: GetLocationResponse = await self._app_client.GetLocation(request, metadata=self._metadata)
         return response.location
 
     async def update_location(self, location_id: str, name: Optional[str] = None, parent_location_id: Optional[str] = None) -> Location:
         """Change the name of a location and/or assign it a new parent location.
 
         Args:
-            location_id (str): ID of the location to update.
+            location_id (str): ID of the location to update. Must be specified.
```

Comment:
We could, but I chose not to just because this method asks for two different location_id's (the one to update, and the optional one to assign as the new parent). We totally could allow it to default. Do you want me to? I think it makes sense either way.

Reply:
Oh nah, that's totally reasonable. Fine to leave as is :)